### ISSUE TITLE
KeyboardInterrupt ignored during template rendering

### ISSUE THREAD

**Issue by jonapich** [pallets/jinja#971]
My console output came out like this: (jinja 2.10.1)

```
Exception ignored in: <generator object Node.iter_fields at 0x7f14c4cd36d0>
Traceback (most recent call last):
  File "~/.virtualenvs/py38dev/lib/python3.8/site-packages/jinja2/nodes.py", line 160, in iter_fields
    yield name, getattr(self, name)
KeyboardInterrupt: 
```
It was rendering a template when I hit the "Stop"button in PyCharm (no debugger attached). The KeyboardInterrupt exception was not propagated and my execution loop went on like nothing happened.

I'm using this syntax currently: `return Template(template).render(**stuff)`

**Comment by EtiennePelletier** [comment]
Salut Jonathan,

I've tried reproducing it with a minimal jinja2 example and was not able to. When I sent the KeyboardInterrupt signal to my program, it was properly caught and the program stopped.
I was using Python 3.7.3 in Linux.

Perhaps the problem was due to the environment difference... Could you try to run the following snippet with py38dev and PyCharm and see if you can reproduce your issue?
Basically send a KeyboardInterrupt in the first 5 seconds of the program's execution and see if it is not propagated properly.
```python
#!/usr/bin/env python

import time
import jinja2

def wait(x):
    time.sleep(x)
    return x ** 2

temp = jinja2.Template("{{ wait(5) }}")

temp.globals['wait'] = wait

print(temp.render())
```

😄
